substratevm/src/com.oracle.svm.core/src/com/oracle/svm/core/JavaMainWrapper.java

```diff
@@ -133,6 +135,12 @@ public static int runCore(int paramArgc, CCharPointerPointer paramArgv) {
         int exitCode;
         try {
             if (SubstrateOptions.ParseRuntimeOptions.getValue()) {
+                if (SubstrateUtil.isPosix()) {
```

Comment:
Why does that not work the same way on Windows too?

Reply:
I will have a look what the JVM does on Windows and make --install-exit-handlers behave the same. Then this is no more posix-only.